src/toil/job.py

```diff
@@ -1496,6 +1531,56 @@ def run(self, fileStore):
         rValue = userFunction(*((self,) + tuple(self._args)), **self._kwargs)
         return rValue
 
+
+class PromisedRequirementFunctionWrappingJob(FunctionWrappingJob):
+    """
+    Adds child function using parent function parameters and fulfilled Promise values.
+
+    (see :class:`toil.job.FunctionWrappingJob` and :class:`toil.job.Job`)
+    """
+
+    def __init__(self, userFunction, *args, **kwargs):
+
+        self.promised_requirement = {}
+
+        # Reset any PromisedRequirement instances to default values until the promise can be fulfilled.
```

Comment:
Don't like the use of the word default here. These are the requirements needed for the execution of the intermediate job. There is nothing "default" about this.